src/metabase/sync_database/sync.clj

```diff
@@ -275,3 +279,25 @@
     (set-fk-relationships! database)
     ;; HACK! we can't sync the _metabase_metadata table until all the "Raw" Tables/Columns are backed
     (maybe-sync-metabase-metadata-table! database raw-tables)))
+
+(defn future-sync-and-analyze-table
```

Comment:
I think this function should probably be called something like `sync-and-analyze-table-async!`. 

1.  It has side-effects -- calling it twice is not the same as calling it once. (Calling it twice would mean a table would get synced twice.) https://github.com/bbatsov/clojure-style-guide#changing-state-fns-with-exclamation-mark
2.  The fact that it uses a `future` is an implementation detail of it being asynchronous. Saying `async` instead of `future` hides those implementation details and leaves open the door we might implement it in some other way in the future without having to change the name of it. Also I think people less familiar with Clojure would not necessarily know what `future-` meant right away but `-async` should be clear to just about everyone